src/clj_holmes/rules/loader/compose.clj

```diff
@@ -0,0 +1,27 @@
+(ns clj-holmes.rules.loader.compose
+  (:require [clj-holmes.rules.loader.builder :as rules.builder]
+            [clojure.walk :as walk]))
+
+(defn ^:private condition-fn-by-condition
+  "Returns a function to validate the rule patterns results."
+  [condition]
+  (case condition
+    :and (fn [& elements]
```

Comment:
Is there a reason that the `:pattern` case here returns a function that takes multiple arguments and `:pattern-not` returns a function of only argument? Both cases are always going to run with only one argument.